src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1136,6 +1136,14 @@ def forward(self, inputs, targets):
         if self.test_loaders is not None and not isinstance(self.test_loaders, dict):
             raise ValueError("`test_loaders` must be a dictionary mapping dataset names to DataLoaders")
 
+        from super_gradients.training.datasets.collate_fn import maybe_calibrate_dataset_adapter
+
+        # Not mandatory but ensures that - if the dataloader use DatasetAdapterCollateFN - the adapter is properly calibrated before iterating over it.
+        maybe_calibrate_dataset_adapter(dataloader=self.train_loader)
+        maybe_calibrate_dataset_adapter(dataloader=self.valid_loader)
+        for test_loader in self.test_loaders.values():
+            maybe_calibrate_dataset_adapter(dataloader=test_loader)
+
```

Comment:
I don't like this too much, I feel like it should not be called here, it should not be the responsability of the Trainer.
But, it smoothens the user experience - if someone uses the `DatasetAdapterCollateFN` and did not calibrate the adapter, we do it for him.
I'm open to remove it, or to move to somewhere else

Reply:
removed